packages/store/src/internal/unhandled-rxjs-error-callback.ts

```diff
@@ -0,0 +1,40 @@
+import { config } from 'rxjs';
+
+const ɵɵunhandledRxjsErrorCallbacks = new WeakMap<any, VoidFunction>();
+
+const existingHandler = config.onUnhandledError;
+config.onUnhandledError = function (error: any) {
+  const unhandledErrorCallback = ɵɵunhandledRxjsErrorCallbacks.get(error);
+  if (unhandledErrorCallback) {
+    unhandledErrorCallback();
+  } else if (existingHandler) {
+    existingHandler.call(this, error);
+  } else {
+    throw error;
+  }
+};
+
+export function executeUnhandledCallback(error: any) {
+  const unhandledErrorCallback = ɵɵunhandledRxjsErrorCallbacks.get(error);
+  if (unhandledErrorCallback) {
+    unhandledErrorCallback();
+    return true;
+  }
+  return false;
+}
+
+export function assignUnhandledCallback(error: any, callback: VoidFunction) {
+  // Since the error can be essentially anything, we must ensure that we only
+  // handle objects, as weak maps do not allow any other key type besides objects.
+  // The error can also be a string if thrown in the following manner: `throwError('My Error')`.
+  if (error !== null && typeof error === 'object') {
```

Comment:
Do we need a strategy for handling non-object errors?

Reply:
I don't think so. Here's why:

- the ecma 2022 standard specifies the `Error` class which is intended to represent errors
- while other types are technically allowed in `throw` statements, such as `throw some_string`, it's widely regarded as bad practice. This is because it doesn't include stack trace information, making debugging more difficult and less informative